crates/cairo-lint-core/src/fix.rs

```diff
@@ -378,4 +379,135 @@ impl Fixer {
+                }
+            }
+        }
+
+        for statement in loop_expr.body(db).statements(db).elements(db).iter().skip(1) {
+            loop_body.push_str(&format!("{}    {}\n", indent, statement.as_syntax_node().get_text_without_trivia(db)));
+        }
+
+        format!("{}while {} {{\n{}{}}}\n", indent, condition_text, loop_body, indent)
+    }
+
+    fn process_block(&self, db: &dyn SyntaxGroup, block: ExprBlock, indent: &str) -> String {
+        let mut block_body = String::new();
+        for statement in block.statements(db).elements(db) {
+            if !matches!(statement, Statement::Break(_)) {
+                block_body.push_str(&format!(
+                    "{}    {}\n",
+                    indent,
+                    statement.as_syntax_node().get_text_without_trivia(db)
+                ));
+            }
+        }
+        block_body
+    }
+
+    fn process_else_clause(&self, db: &dyn SyntaxGroup, else_clause: ElseClause, indent: &str) -> String {
+        let mut else_body = String::new();
+        match else_clause.else_block_or_if(db) {
+            BlockOrIf::Block(block) => {
+                else_body.push_str(&self.process_block(db, block, indent));
+            }
+            BlockOrIf::If(else_if) => {
+                else_body.push_str(&format!(
+                    "{}else if {} {{\n",
+                    indent,
+                    else_if.condition(db).as_syntax_node().get_text_without_trivia(db)
+                ));
+                else_body.push_str(&self.process_block(db, else_if.if_block(db), indent));
+                else_body.push_str(&format!("{}}}\n", indent));
+            }
+        }
+        else_body
+    }
+
+    fn invert_condition(condition: &str) -> String {
+        if condition.contains("&&") {
+            condition
+                .split("&&")
+                .map(|part| Self::invert_simple_condition(part.trim()))
+                .collect::<Vec<_>>()
+                .join(" || ")
+        } else if condition.contains("||") {
+            condition
+                .split("||")
+                .map(|part| Self::invert_simple_condition(part.trim()))
+                .collect::<Vec<_>>()
+                .join(" && ")
+        } else {
+            Self::invert_simple_condition(condition)
+        }
+    }
+
+    fn invert_simple_condition(condition: &str) -> String {
+        if condition.contains(">=") {
+            condition.replace(">=", "<")
+        } else if condition.contains("<=") {
+            condition.replace("<=", ">")
+        } else if condition.contains(">") {
+            condition.replace(">", "<=")
+        } else if condition.contains("<") {
+            condition.replace("<", ">=")
+        } else if condition.contains("==") {
+            condition.replace("==", "!=")
+        } else if condition.contains("!=") {
+            condition.replace("!=", "==")
+        } else {
+            format!("!({})", condition)
+        }
+    }
```

Comment:
can you create a helper module in the fix module? all those functions will for sure be useful to someone else. Also can you check if there is any overlap with the function `indent_snippet` ?